### ISSUE TITLE
Java 9: Please add Automatic-Module-Name attribute to META-INF/MANIFEST.MF

### ISSUE THREAD

**Issue by remal** [testcontainers/testcontainers-java#1523]
I'm using `org.testcontainers:postgresql:1.11.3` dependency and it has no `Automatic-Module-Name` manifest attribute. It leads to module name conflict, as module name for the dependency is just `postgres`.

Please add this attribute to all published artifacts. For postgresql it can be smth like `testcontainers.postgresql`.

**Comment by kiview** [comment]
Thanks @remal, you are 100% right that this is a PITA and we should fix this soon!

**Comment by heruan** [comment]
This is really needed, there's a conflict with `org.testcontainers:postgresql` and `org.postgresql:postgresql` which inhibits the use of both modules!

**Comment by rnorth** [comment]
Agreed. We'd happily accept a PR for this, if anyone wants to speed this up.

**Comment by remal** [comment]
I think that #1524 is much more important. Modules names will be useful if projects can't be compiled.

**Comment by heruan** [comment]
@rnorth would the `testcontainers.`-prefix be acceptable? So modules will have names like:

- `testcontainers.jdbc`
- `testcontainers.mysql`
- `testcontainers.postgresql`

**Comment by heruan** [comment]
@rnorth I've issued the PR to speed this up, hope the module names are fine otherwise I'll change them!

**Comment by heruan** [comment]
BTW @remal it totally right, fixing module names still does permit to use them since different modules share the same package name `org.testcontainers.container` which is not permitted by the spec:

> _Non-interference_ — The Java compiler, virtual machine, and run-time system must ensure that modules that contain packages of the same name do not interfere with each other. If two distinct modules contain packages of the same name then, from the perspective of each module, all of the types and members in that package are defined only by that module. Code in that package in one module must not be able to access package-private types or members in that package in the other module.

— http://openjdk.java.net/projects/jigsaw/spec/reqs/#non-interference

So also #1524 should be fixed as soon as possible.

**Comment by A248** [comment]
I'm wondering about the status of this since it is a blocker for any modular project.

The two problems seem to be that testcontainers contains split packages and does not declare a stable module name. Declaring a module name is not very useful without resolving the split packages, of course. Renaming this issue might be useful since it is not merely adding an automatic module name.

By the way, there are a number of duplicates of this issue. Maybe you would like to consolidate them:

* https://github.com/testcontainers/testcontainers-java/issues/521
* https://github.com/testcontainers/testcontainers-java/issues/1524
* https://github.com/testcontainers/testcontainers-java/issues/1917

> Agreed. We'd happily accept a PR for this, if anyone wants to speed this up.

As I understand it, fixing the split packages will require API changes. Is the 2.0 version already in development?

**Comment by shedfull** [comment]
Is there any news about this issue? As it is currently not possible to use testcontainers in a modular project, which is a real shame. Has anyone worked on this? I'd be happy to help
